map_model/src/make/turns.rs

```diff
@@ -6,12 +6,12 @@ use lyon::geom::{CubicBezierSegment, Point, QuadraticBezierSegment};
 use geom::{PolyLine, Pt2D};
 
 use crate::{
-    map::turn_type_from_road_geom, Intersection, Lane, LaneID, LaneType, Map, RoadID, Turn, TurnID,
+    map::turn_type_from_road_geom, Intersection, IntersectionID, Lane, LaneID, LaneType, Map, RoadID, Turn, TurnID,
     TurnType,
 };
 
 /// Generate all driving and walking turns at an intersection, accounting for OSM turn restrictions.
-pub fn make_all_turns(map: &Map, i: &Intersection) -> Vec<Turn> {
+pub fn make_all_turns(map: &Map, i: &Intersection) -> (Vec<Turn>, Vec<Turn>) {
```

Comment:
This now returns (allowed, banned) turns, right? Could we update the docstring?

Reply:
I have reverted this change (https://github.com/a-b-street/abstreet/commit/e3dfbd862a1b10e506d25611d1eaaa8d71bf7724). The 'banned' Vec was not used anywhere, and I was not confident the results were correct.

Also, since writing up the definitions, I think generating three lists `(permitted, opposing_oneway, restricted_turn)` would more useful (See comment below):
https://github.com/a-b-street/abstreet/blob/e3dfbd862a1b10e506d25611d1eaaa8d71bf7724/apps/ltn/src/logic/turn_restrictions.rs#L166